In [1]:
%matplotlib inline
import pandas as pd
import matplotlib.pyplot as plt
import math
import numpy as np

import altair as alt
from vega_datasets import data

olympics = pd.read_csv('data/athlete_events.csv') 

In [59]:
olympics.shape

(271116, 15)

There are 271116 rows and 15 columns. The unit of analysis is event per athlete per game. 

In [3]:
olympics.head(10)

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   
5   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   
6   5  Christine Jacoba Aaftink   F  25.0   185.0    82.0     Netherlands   
7   5  Christine Jacoba Aaftink   F  25.0   185.0    82.0     Netherlands   
8   5  Christine Jacoba Aaftink   F  27.0   185.0    82.0     Netherlands   
9   5  Christine Jacoba Aaftink   F  27.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season         City          Sport  \
0  CHN  1992 Summer  1992  Summer    Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer       London           Judo   
2  DEN  1920 Summer  1920  Summer    Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer        Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter      Calgary  Speed Skating   
5  NED  1988 Winter  1988  Winter      Calgary  Speed Skating   
6  NED  1992 Winter  1992  Winter  Albertville  Speed Skating   
7  NED  1992 Winter  1992  Winter  Albertville  Speed Skating   
8  NED  1994 Winter  1994  Winter  Lillehammer  Speed Skating   
9  NED  1994 Winter  1994  Winter  Lillehammer  Speed Skating   

                                Event Medal  
0         Basketball Men's Basketball   NaN  
1        Judo Men's Extra-Lightweight   NaN  
2             Football Men's Football   NaN  
3         Tug-Of-War Men's Tug-Of-War  Gold  
4    Speed Skating Women's 500 metres   NaN  
5  Speed Skating Women's 1,000 metres   NaN  
6    Speed Skating Women's 500 metres   NaN  
7  Speed Skating Women's 1,000 metres   NaN  
8    Speed Skating Women's 500 metres   NaN  
9  Speed Skating Women's 1,000 metres   NaN

**What was the first year recorded in this dataset, and what was the most recent?**

In [60]:
olympics.min()['Year']

1896

In [5]:
olympics.max()['Year']

2016

This data includes Olympics from 1896 through 2016. 

**How many Olympic games have there been?** 

In [6]:
olympics.nunique()['Games']

51

**How many unique Olympic athletes have there been?**

In [7]:
olympics.nunique()["ID"]

135571

There have been 135,571 athletes that have participated in the Olympics since 1896.

**How many athletes have been male, how many have been women?**

In [2]:
male = olympics['Sex'] == "M"
male_athletes= olympics[male]

In [7]:
male_athletes.nunique()["ID"]

101590

101,590 athletes have been male. 

In [3]:
female = olympics['Sex'] == "F"
female_athletes= olympics[female]

In [8]:
female_athletes.nunique()["ID"]

33981

33,981 have been women. 

There have been much less women athletes than men athletes who have participated in the Olympics. 
**WHY?** 

**What is the first year in which a woman athlete participated in the games?**

In [38]:
female_athletes["Year"].min()

1900

The first year a woman participated in the Olympics was in 1900. 

**Graph participation of women versus men over time** 

In [11]:
female_by_year = female_athletes.groupby(["Year"])["Sex"].value_counts()
print(female_by_year)

Year  Sex
1900  F        33
1904  F        16
1906  F        11
1908  F        47
1912  F        87
1920  F       134
1924  F       261
1928  F       437
1932  F       369
1936  F       549
1948  F       761
1952  F      1682
1956  F      1139
1960  F      1730
1964  F      1752
1968  F      2193
1972  F      2608
1976  F      2606
1980  F      2186
1984  F      2983
1988  F      4223
1992  F      5178
1994  F      1105
1996  F      5008
1998  F      1384
2000  F      5431
2002  F      1582
2004  F      5546
2006  F      1757
2008  F      5816
2010  F      1847
2012  F      5815
2014  F      2023
2016  F      6223
Name: Sex, dtype: int64


In [47]:
male_by_year = male_athletes.groupby(["Year"])["Sex"].value_counts()
print(male_by_year)

Year  Sex
1896  M        380
1900  M       1903
1904  M       1285
1906  M       1722
1908  M       3054
1912  M       3953
1920  M       4158
1924  M       5432
1928  M       5137
1932  M       2952
1936  M       6852
1948  M       6719
1952  M       7676
1956  M       5295
1960  M       7505
1964  M       7728
1968  M       8286
1972  M       9351
1976  M       7896
1980  M       6751
1984  M       8605
1988  M      10453
1992  M      11235
1994  M       2055
1996  M       8772
1998  M       2221
2000  M       8390
2002  M       2527
2004  M       7897
2006  M       2625
2008  M       7786
2010  M       2555
2012  M       7105
2014  M       2868
2016  M       7465
Name: Sex, dtype: int64


DO THIS INTERPRETATION 
The is a difference between male and female participation. The alternating period on the right hand part of the graph shows the difference in participation between summer and winter games. There is greater participation of men in summer